src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,262 @@
+        printSign();
+    }
+
+    public static void addTodo(String userCommand) {
+        //add exception
+        try {
+            int contentStart = 5;
+            String description = userCommand.substring(contentStart);
+            tasks.add(new Todo(description));
+            taskNum++;
+            printTotalNumOfTasks(tasks.get(taskNum - 1), taskNum);
+        }
+        catch (StringIndexOutOfBoundsException e) {
+            printSign();
+            System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+            printSign();
+        }
+    }
+
+    public static void addEvent(String userCommand) {
+        //add exception
+        try {
+            int contentStart = 6;
+            int contentEnd = userCommand.indexOf("/at") - 1;
+            int atStart = userCommand.indexOf("/at") + 4;
+            String description = userCommand.substring(contentStart, contentEnd);
+            String at = userCommand.substring(atStart);
+            tasks.add(new Event(description, at));
+            taskNum++;
+            printTotalNumOfTasks(tasks.get(taskNum - 1), taskNum);
+        }
+        catch (StringIndexOutOfBoundsException e) {
+            printSign();
+            System.out.println("☹ OOPS!!! The description of a event cannot be empty.");
+            printSign();
+        }
+    }
+
+    public static void addDeadline(String userCommand) {
+        //add exception
+        try {
+            int contentStart = 9;
+            int contentEnd = userCommand.indexOf("/by") - 1;
+            int byStart = userCommand.indexOf("/by") + 4;
+            String description = userCommand.substring(contentStart, contentEnd);
+            String by = userCommand.substring(byStart);
+            tasks.add(new Deadline(description, by));
+            taskNum++;
+            printTotalNumOfTasks(tasks.get(taskNum - 1), taskNum);
+        }
+        catch (StringIndexOutOfBoundsException e) {
+            printSign();
+            System.out.println("☹ OOPS!!! The description of a deadline cannot be empty.");
+            printSign();
+        }
+    }
+
+    public static void deleteTask(String userCommand) {
+        try {
+            int contentStart = 7;
+            int id = Integer.parseInt(userCommand.substring(contentStart));
+            System.out.println("Noted. I've removed this task: ");
+            System.out.println(tasks.get(id - 1).toString());
+            tasks.remove(id - 1);
+            taskNum--;
+            System.out.println("Now you have " + taskNum + " tasks in the list.");
+            printSign();
+        }
+        catch (StringIndexOutOfBoundsException e) {
+            printSign();
+            System.out.println("☹ OOPS!!! The index of the task to be deleted cannot be empty.");
+            printSign();
+        }
+    }
+
+    public static void taskDone(Task args) {
+        // Mark the relevant task as "done", and print out a line indicates that the task is marked as done
```

Comment:
Good job with providing useful comments to describe the methods here. However, you could format these as method headers (or [javadocs](https://se-education.org/guides/conventions/java/basic.html#comments)) instead.